Jenkinsfile_nightly

```diff
@@ -0,0 +1,195 @@
+            keepAll              : true,
+            reportDir            : "application/build/reports/tests/test/",
+            reportFiles          : 'index.html',
+            reportName           : 'Unit Test Report'
+        ])
+
+        publishHTML([
+            allowMissing         : true,
+            alwaysLinkToLastBuild: true,
+            keepAll              : true,
+            reportDir            : "application/build/reports/checkstyle/",
+            reportFiles          : 'main.html',
+            reportName           : 'Checkstyle Main Report'
+        ])
+        publishHTML([
+            allowMissing         : true,
+            alwaysLinkToLastBuild: true,
+            keepAll              : true,
+            reportDir            : "application/build/reports/checkstyle/",
+            reportFiles          : 'test.html',
+            reportName           : 'Checkstyle Test Report'
+        ])
+        publishHTML([
+            allowMissing         : true,
+            alwaysLinkToLastBuild: true,
+            keepAll              : true,
+            reportDir            : "application/build/reports/pmd/",
+            reportFiles          : 'main.html',
+            reportName           : 'PMD Main Report'
+        ])
+        publishHTML([
+            allowMissing         : true,
+            alwaysLinkToLastBuild: true,
+            keepAll              : true,
+            reportDir            : "application/build/reports/pmd/",
+            reportFiles          : 'test.html',
+            reportName           : 'PMD Test Report'
+        ])
+
+        try {
+            sh './gradlew jacocoTestReport --info'
+        }finally {
+            jacoco(execPattern: 'application/build/jacoco/test.exec', buildOverBuild: false,
+                exclusionPattern: '**/test/*, ' +
+                    '**/uk/gov/hmcts/reform/em/annotation/EmAnnotationApp.java,'+
+                    '**/uk/gov/hmcts/reform/em/annotation/hateos/*,'+
+                    '**/uk/gov/hmcts/reform/em/annotation/exception/*,'+
+                    '**/uk/gov/hmcts/reform/em/annotation/domain/*,'+
+                    '**/uk/gov/hmcts/reform/em/annotation/commandobject/*,'+
+                    '**/uk/gov/hmcts/reform/em/annotation/hibernate/*,'+
+                    '**/uk/gov/hmcts/reform/em/annotation/config/**/*,'+
+                    '**/uk/gov/hmcts/reform/em/annotation/errorhandler/*,'+
+                    '**/uk/gov/hmcts/reform/em/annotation/repository/RepositoryFinder.java')
+            publishHTML([
+                allowMissing         : true,
+                alwaysLinkToLastBuild: true,
+                keepAll              : true,
+                reportDir            : "application/build/reports/jacoco/test/html/",
+                reportFiles          : 'index.html',
+                reportName           : 'Jacoco Coverage Report'
+            ])
+        }
+    }
+
+    after('smoketest:demo') {
```

Comment:
Everything starting from this line can be removed as there's no deployments, smoke or functional tests.